src/jvm/test/org/jetbrains/kotlinx/lincheck_test/util/TestUtils.kt

```diff
@@ -24,6 +29,13 @@ import java.io.File
 import java.util.concurrent.ThreadFactory
 import java.util.concurrent.atomic.AtomicInteger
 
+internal inline fun withLincheckJavaAgentAndLincheckTestLock(
+    instrumentationMode: InstrumentationMode,
+    block: () -> Unit
+): Unit = LINCHECK_TEST_LOCK.withLock {
+    withLincheckJavaAgent(instrumentationMode, block)
+}
```

Comment:
BTW, we can move it to `Lincheck.kt`, rename to something like `lincheckTest` or `withLincheckTestContext` and use it both in tests and Lincheck itself.